easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -85,205 +80,220 @@ public static boolean hasPermissions(@NonNull Context context, @NonNull String..
-            } else {
-                // Revert to AlertDialog
-                showRationaleAlertDialog(object, rationale, positiveButton, negativeButton,
-                        requestCode, perms);
-            }
+        if (shouldShowRationale(activity, perms)) {
+            showRationaleDialogFragment(
+                    activity.getFragmentManager(),
+                    rationale,
+                    positiveButton,
+                    negativeButton,
+                    requestCode,
+                    perms);
         } else {
-            executePermissionsRequest(object, perms, requestCode);
+            ActivityCompat.requestPermissions(activity, perms, requestCode);
         }
     }
 
     /**
-     * Show a {@link RationaleDialogFragmentCompat} explaining permission request rationale.
-     */
-    @RequiresApi(Build.VERSION_CODES.HONEYCOMB)
-    private static void showRationaleDialogFragmentCompat(
-            @NonNull final android.support.v4.app.FragmentManager fragmentManager,
-            @NonNull String rationale, @StringRes int positiveButton, @StringRes int negativeButton,
-            final int requestCode, @NonNull final String... perms) {
-
-        RationaleDialogFragmentCompat fragment = RationaleDialogFragmentCompat
-                .newInstance(positiveButton, negativeButton, rationale, requestCode, perms);
-        fragment.show(fragmentManager, DIALOG_TAG);
-    }
-
-    /**
-     * Show a {@link RationaleDialogFragment} explaining permission request rationale.
+     * Request a set of permissions, showing rationale if the system requests it.
+     *
+     * @param fragment    Fragment requesting permissions. Should implement {@link
+     *                    android.support.v4.app.ActivityCompat.OnRequestPermissionsResultCallback}
+     *                    or {@code android.support.v13.app.FragmentCompat.OnRequestPermissionsResultCallback}
+     * @param rationale   a message explaining why the application needs this set of permissions,
+     *                    will be displayed if the user rejects the request the first time.
+     * @param requestCode request code to track this request, must be < 256.
+     * @param perms       a set of permissions to be requested.
+     * @see android.Manifest.permission
      */
-    @RequiresApi(api = Build.VERSION_CODES.HONEYCOMB)
-    private static void showRationaleDialogFragment(
-            @NonNull final android.app.FragmentManager fragmentManager,
-            @NonNull String rationale, @StringRes int positiveButton, @StringRes int negativeButton,
-            final int requestCode, @NonNull final String... perms) {
-
-        RationaleDialogFragment fragment = RationaleDialogFragment
-                .newInstance(positiveButton, negativeButton, rationale, requestCode, perms);
-        fragment.show(fragmentManager, DIALOG_TAG);
+    public static void requestPermissions(@NonNull Fragment fragment,
```

Comment:
I agree with you that this can be confusing so I'm open to adding the full qualifier, but for the moment I just made everything a link in 941c08f. I really hope that's good enough because it kills me to add the full qualifier.